src/main/java/seedu/address/model/client/policy/UniquePolicyList.java

```diff
@@ -0,0 +1,141 @@
+package seedu.address.model.client.policy;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Iterator;
+import java.util.List;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import seedu.address.model.client.exceptions.DuplicatePolicyException;
+import seedu.address.model.client.exceptions.PolicyNotFoundException;
+
+
+/**
+ * Represents a list of policies that enforces no two policies in the list have the same identity.
+ * A policy is considered the same as another policy in the list if they have the same identity.
+ * Identity is defined by {@code Policy#isSamePolicy(Policy)}.
+ *
+ * Supports a minimal set of list operations.
+ *
+ * @see Policy#isSamePolicy(Policy)
+ */
+public class UniquePolicyList implements Iterable<Policy> {
```

Comment:
How do I determine if the `UniquePolicyList` is empty? `Iterable` does not have an `isEmpty()` method.